pkg/workceptor/workceptor.go

```diff
@@ -689,6 +689,31 @@ func (w *Workceptor) CancelUnit(unitID string) error {
 		return fmt.Errorf("unknown work unit %s", unitID)
 	}
 	w.activeUnitsLock.RUnlock()
+	remoteUnitID := unit.status.RemoteUnitID
+	if remoteUnitID != "" {
+		remoteNodeID := unit.status.Node
+		conn, reader, err := w.connectToRemote(remoteNodeID)
+		if err != nil {
+			logger.Error("Connection failed to %s: %s\n", remoteNodeID, err)
+			return err
+		}
+		_, err = conn.Write([]byte(fmt.Sprintf("work cancel %s\n", remoteUnitID)))
+		if err != nil {
+			logger.Error("Write error sending to %s: %s\n", remoteNodeID, err)
+			return err
+		}
+		response, err := reader.ReadString('\n')
+		if err != nil {
+			logger.Error("Read error reading from %s: %s\n", remoteNodeID, err)
+			return err
+		}
+		cancelledRegex := regexp.MustCompile(fmt.Sprintf("cancelled.*%s", remoteUnitID))
```

Comment:
maybe this is not the proper way to check. We could monitor the remote work status.State instead

Reply:
Yes ^^